src/utils/download.ts

```diff
@@ -0,0 +1,54 @@
+import * as fs from 'fs';
+import path from 'path';
+import axios from 'axios';
+
+type DownloadProgressCallback = (downloadProgress: number) => void;
+type DownloadResult = { pathToFile: string; contentType: string };
+
+/**
+ * Downloads file from remote HTTP[S] host and puts its contents to the
+ * specified location.
+ * @param url URL of the file to download.
+ * @param destinationPath Path to the destination file.
+ * @param destinationFileName The file name.
+ */
+
+export async function download(
+	url: string,
+	destinationPath: string,
+	destinationFileName: string,
+	downloadProgressCallback?: DownloadProgressCallback
+): Promise<DownloadResult> {
+	const pathToFile = path.join(destinationPath, destinationFileName);
+	const writer = fs.createWriteStream(pathToFile);
+	try {
+		const { data, headers } = await axios({
+			method: 'get',
+			url: url,
+			responseType: 'stream'
+		});
+		const totalLength = headers['content-length'];
+		const contentType = headers['content-type'];
+		let downloadedLength = 0;
+		data.on('data', (chunk: string | unknown[]) => {
+			downloadedLength += chunk.length;
+			downloadProgressCallback && downloadProgressCallback((downloadedLength / totalLength) * 100);
```

Comment:
nits:

```suggestion
			const downloadProgressPct = (downloadedLength / totalLength) * 100;
			downloadProgressCallback?.(downloadProgressPct);
```

Reply:
Going further, perhaps overkill but I never trust division of a variable:
```
const downloadProgressPct = (totalLength > 0) ? (downloadedLength / totalLength) * 100 : 0;
```